src/gameplayv2/resistance/entities/Entity.ts

```diff
@@ -0,0 +1,37 @@
+class Entity{
+    static nextId = 0;
+    protected components: any[];
+    public role: string;
+    public alliance: string;
+    public description: string;
+
+    constructor() {
+        this.components = [];
+        this.role;
+        this.alliance;
+        this.description;
+    }
+
+    public addComponent(component: any): void {
+        this.components.push(component);
+    }
+
+    public addComponents(...components: any[]): void {
+        components.forEach((component) => this.addComponent(component));
+    }
```

Comment:
Let's remove this. I don't see a strong need for this at the moment? Usually when we add components they'll be unique ones anyway, and not too many.